src/secure-value.ts

```diff
@@ -102,16 +101,32 @@ export const serializedSecureEnvSourceText = (function secureEnvFactory(rawEnv:
         return obj === undefined;
     }
 
+    function isNull(obj: any): obj is null {
+        return obj === null;
+    }
+
     function isFunction(obj: any): obj is Function {
         return typeof obj === 'function';
     }
 
-    function isNullish(obj: any): obj is null {
-        // eslint-disable-next-line eqeqeq
-        return obj == null;
+    function isNullOrUndefined(obj: any): obj is (null | undefined) {
+        return isNull(obj) || isUndefined(obj);
     }
 
     function getSecureValue(raw: RawValue): SecureValue {
+        if (isNullOrUndefined(raw)) {
+            return raw as SecureValue;
+        }
+        const t = typeof raw;
```

Comment:
i believe caching the typeof value like this can hurt performance; i've been pushed to avoid this pattern in the es6-shim for that reason (but don't have benchmarks). iow, the faster pattern (at the time, at least) was to always repeat `typeof raw === ` inline, because that's what engines optimized for.

Reply:
@ljharb I can see why! let me run some benchmarks, and update that in another PR since we do it in multiple places!